services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/service/impl/DhcpService.java

```diff
@@ -87,14 +89,6 @@ public void buildDhcpStates(NetworkConfiguration networkConfig, UnicastGoalState
 
                 DHCP.DHCPState dhcpState = dhcpStateBuilder.build();
                 unicastGoalState.getGoalStateBuilder().putDhcpStates(dhcpState.getConfiguration().getId(), dhcpState);
-
-                Goalstate.ResourceIdType dhcpResourceIdType = Goalstate.ResourceIdType.newBuilder()
-                        .setType(Common.ResourceType.DHCP)
-                        .setId(dhcpState.getConfiguration().getId())
-                        .build();
-                Goalstate.HostResources.Builder hostResourceBuilder = Goalstate.HostResources.newBuilder();
-                hostResourceBuilder.addResources(dhcpResourceIdType);
-                unicastGoalState.getGoalStateBuilder().putHostResources(unicastGoalState.getHostIp(), hostResourceBuilder.build());
```

Comment:
@DavidLiu506 Do we need to re-apply the same logics, or this is completely unnecessary?

Reply:
I saw the GS builders for different types were removed in multiple Service.java under service/impl folder.

Reply:
Yes, remove code is not correct. resouceId replaced previous resouceid.

For example:
There are 3 subnets and resource id is 1, 2, 3

unicastGoalState.getGoalStateBuilder().putHostResources(10.213.43.166, [1, 2, 3]);
This will add subnets to goal state.
Then add a port state:
port resource id is 1
unicastGoalState.getGoalStateBuilder().putHostResources(10.213.43.166, [1]);

This will replaced subnet resource id and resourceMap only contains port resource id.